specs/DOMContentLoaded.md

```diff
@@ -0,0 +1,119 @@
+                    )~",
+                    Callback<ICoreWebView2ExecuteScriptCompletedHandler>(
+                        [](HRESULT error, PCWSTR result) -> HRESULT { return S_OK; })
+                        .Get());
+                return S_OK;
+            })
+            .Get(),
+        &m_DOMContentLoadedToken));
+    //! [DOMContentLoaded]
+```
+
+## C#
+```
+webView.CoreWebView2.DOMContentLoaded += (object sender, CoreWebView2DOMContentLoadedEventArgs arg) =>
+{
+    webView.ExecuteScriptAsync("let " +
+                              "content=document.createElement(\"h2\");content.style.color=" +
+                              "'blue';content.textContent= \"This text was added by the " +
+                              "host app\";document.body.appendChild(content);");
+};
+webView.NavigateToString(@"<!DOCTYPE html><h1>DOMContentLoaded sample page</h1><h2>The content below will be added after DOM content is loaded </h2>");
+
+```
+
+# API Notes
+
+See [API Details](#api-details) section below for API reference.
+# API Details
+
+## Win32 C++
+
+```IDL
+interface ICoreWebView2_2;
+interface ICoreWebView2DOMContentLoadedEventArgs;
+interface ICoreWebView2DOMContentLoadedEventHandler;
+
+[uuid(9810c82b-8483-4f1c-b2f4-6244f1010c05), object, pointer_default(unique)]
+interface ICoreWebView2_2 : ICoreWebView2 {
+  /// Add an event handler for the DOMContentLoaded event.
+  /// DOMContentLoaded is raised when the initial html document has been parsed.
+  /// This aligns with the the document's DOMContentLoaded event in html
+  ///
+  /// \snippet ScenarioDOMContentLoaded-Staging.cpp
+  HRESULT add_DOMContentLoaded(
+      [in] ICoreWebView2StagingDOMContentLoadedEventHandler* eventHandler,
+      [out] EventRegistrationToken* token);
+  /// Remove an event handler previously added with add_DOMContentLoaded.
+  HRESULT remove_DOMContentLoaded(
+      [in] EventRegistrationToken token);
+}
+
+/// Event args for the DOMContentLoaded event.
+[uuid(E8BA4206-D6F8-42F1-9A6D-43C8A99C1F39), object, pointer_default(unique)]
+interface ICoreWebView2DOMContentLoadedEventArgs : IUnknown {
+  /// The ID of the navigation.
+  [propget] HRESULT NavigationId([out, retval] UINT64* navigation_id);
+}
+
+/// The caller implements this interface to receive the DOMContentLoaded
+/// event.
+[uuid(1E649181-785D-40B2-B4AE-AFACD3C6B8DD), object, pointer_default(unique)]
+interface ICoreWebView2DOMContentLoadedEventHandler : IUnknown {
+  /// Called to provide the implementer with the event args for the
+  /// corresponding event.
+  HRESULT Invoke(
+      [in] ICoreWebView2* sender,
+      [in] ICoreWebView2DOMContentLoadedEventArgs* args);
+}
+```
+
+## .NET and WinRT
+
+```c#
+namespace Microsoft.Web.WebView2.Core
+{
+    runtimeclass CoreWebView2DOMContentLoadedEventArgs;
+
+    runtimeclass CoreWebView2DOMContentLoadedEventArgs
+    {
+        // CoreWebView2DOMContentLoadedEventArgs
+        UInt64 NavigationId { get; };
+    }
+
+    runtimeclass CoreWebView2
```

Comment:
Do you need to put this new member into a new interface or versioning block? Adding it here will change the signature/IID/vtable of the generated `ICoreWebView2` interface.

Reply:
Yes, it needs to be a new contract section so that it gets a new interface, unless this is intended to be a binary breaking change.

Reply:
This is an issue with all our APIs. I'll follow up.